tpm/commands.go

```diff
@@ -158,6 +158,23 @@ func getCapability(rw io.ReadWriter, cap, subcap uint32) ([]byte, error) {
 	return b, nil
 }
 
+// nvDefineSpace allocates space in NVRAM
+func nvDefineSpace(rw io.ReadWriter, nvData NVDataPublic, enc digest, ca *commandAuth) ([]byte, *responseAuth, uint32, error) {
+	var b tpmutil.U32Bytes
+	var ra responseAuth
+	in := []interface{}{nvData, enc}
+	if ca != nil {
+		in = append(in, ca)
+	}
+	out := []interface{}{&b, &ra}
```

Comment:
I think your issue is here. You are expecting a `U32Bytes` followed by the `responseAuth`, which would correspond to these outgoing operands:

 - uint32 (for the size of the data section)
 - []byte (for the variable data section)
 - nonceEven (part of `responseAuth`)
 - nonceOdd (part of `responseAuth`)
 - bool continueAuthSession (part of `responseAuth`)
 - authdata ownerAuth (part of `responseAuth` I think?)

My read of nvDefineSpace outgoing operands is that is only has the `responseAuth` outgoing parameters. As such, will it work if you do: `out := []interface{}{&ra}` and get rid of the `b` local?

Reply:
Yes indeed good sir. Thank you very much!